gui/default/syncthing/core/syncthingController.js

```diff
@@ -1343,16 +1358,30 @@ angular.module('syncthing.core')
         $scope.directoryList = [];
 
         $scope.$watch('currentFolder.path', function (newvalue) {
-            if (newvalue && newvalue.trim().charAt(0) === '~') {
-                $scope.currentFolder.path = $scope.system.tilde + newvalue.trim().substring(1);
+            if (!(newvalue)) {
+                return;
             }
+            $scope.currentFolder.path = expandTilde(newvalue);
             $http.get(urlbase + '/system/browse', {
                 params: { current: newvalue }
             }).success(function (data) {
                 $scope.directoryList = data;
             }).error($scope.emitHTTPError);
         });
 
+        $scope.$watch('currentFolder.label', function (newvalue) {
+            if (!$scope.config.options || !$scope.config.options.defaultFolderPath || $scope.currentFolder.path) {
+                return;
+            }
+            var path = $scope.config.options.defaultFolderPath;
+            if (newvalue) {
+                path = pathJoin(path, newvalue);
+            } else {
+                path = pathJoin(path, $scope.currentFolder.id);
+            }
+            $('#folderPath').prop("value", path);
```

Comment:
I did it with two watches instead of watchGroup. Updating angularJS up to 1.5 seemed to be fine, 1.6 broke spectacularly - I probably shouldn't touch that just as an aside.